ps.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"text/tabwriter"
+
+	"github.com/hyperhq/runv/containerd/api/grpc/types"
+	"github.com/urfave/cli"
+	netcontext "golang.org/x/net/context"
+)
+
+var psCommand = cli.Command{
+	Name:      "ps",
+	Usage:     "ps displays the processes running inside a container",
+	ArgsUsage: `<container-id> [ps options]`, Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "format, f",
+			Value: "table",
+			Usage: `select one of: ` + formatOptions,
+		},
+	},
+	Action: func(context *cli.Context) error {
+		container := context.Args().First()
+		if container == "" {
+			return cli.NewExitError("container id cannot be empty", -1)
+		}
+		c, err := getContainerApi(context, container)
+		if err != nil {
+			return cli.NewExitError(fmt.Sprintf("can't access container, %v", err), -1)
+		}
+
+		switch context.String("format") {
+		case "table":
+			w := tabwriter.NewWriter(os.Stdout, 12, 1, 3, ' ', 0)
+			fmt.Fprint(w, "PROCESS\tCMD\n")
+			// we are limited by the containerd interface for now
+			for _, p := range c.Processes {
+				fmt.Fprintf(w, "%s\t%s\n",
+					p.Pid,
+					p.Args)
+			}
+			if err := w.Flush(); err != nil {
+				fatal(err)
+			}
+		case "json":
+			pids := make([]string, 0)
+			for _, p := range c.Processes {
+				pids = append(pids, p.Pid)
+			}
+
+			data, err := json.Marshal(pids)
+			if err != nil {
+				fatal(err)
+			}
+			os.Stdout.Write(data)
```

Comment:
I will merge it although the problems your reported are not addressed... You can fix it later.